### ISSUE TITLE
win_reg_stat fails when registry key is in HKEY_USERS\.DEFAULT hive

### ISSUE THREAD

**Issue by mgoleary** [ansible/ansible#59337]
##### SUMMARY
win_reg_stat module fails when running against keys in the HKEY_USERS\.DEFAULT hive. All other hives (HKEY_CURRENT_USER, HKEY_LOCAL_MACHINE\*, etc) appear to be working fine, however. 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
win_reg_stat module

##### ANSIBLE VERSION

```paste below
ansible 2.8.1
  config file = /home/admin/ansible-src/com/winrmtesting/ansible.cfg
  configured module search path = [u'/home/admin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /users/ansible/lib/python2.7/site-packages/ansible
  executable location = /users/ansible/bin/ansible
  python version = 2.7.5 (default, Jun 11 2019, 12:19:05) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]

```

##### CONFIGURATION

```paste below
DEFAULT_CALLBACK_WHITELIST(/home/admin/ansible-src/com/winrmtesting/ansible.cfg) = [u'profile_tasks']
DEFAULT_HOST_LIST(/home/admin/ansible-src/com/winrmtesting/ansible.cfg) = [u'/home/admin/ansible-src/com/winrmtesting/inventory/hosts']
HOST_KEY_CHECKING(env: ANSIBLE_HOST_KEY_CHECKING) = False

```

##### OS / ENVIRONMENT
target OS: Windows Server 2016 Datacenter

##### STEPS TO REPRODUCE

Run the win_reg_stat module against any key in the HKEY_USERS\.DEFAULT hive

```yaml
  - name: HKU:\.DEFAULT\Software\Policies\Microsoft\Windows\CurrentVersion\PushNotifications 
    win_reg_stat:
      path: 'HKU:\.DEFAULT\Software\Policies\Microsoft\Windows\CurrentVersion\PushNotifications'
      name: NoToastApplicationNotificationOnLockScreen
    register: pushnotifications_regkey_stat

```

##### EXPECTED RESULTS

ok: [172.17.176.59] => {
    "pushnotifications_regkey_stat": {
        "changed": false, 
        "exists": true, 
        "failed": false, 
        "raw_value": 1, 
        "type": "REG_DWORD", 
        "value": 1
    }
}

##### ACTUAL RESULTS

```paste below
The full traceback is:
You cannot call a method on a null-valued expression.
At line:122 char:5
+     $registry_hive.Dispose()
+     ~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvokeMethodOnNull

ScriptStackTrace:
at <ScriptBlock>, <No file>: line 122

fatal: [172.17.176.59]: FAILED! => {
    "changed": false, 
    "msg": "Unhandled exception while executing module: You cannot call a method on a null-valued expression."
}

```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/windows/win_reg_stat.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_reg_stat.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @ShachafGoldstein @if-meaton @jborean93 @jhawkesworth @nitzmahone
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mgoleary** [comment]
https://github.com/ansible/ansible/pull/53305/commits/e41312db3ca163a29609e1806f2e14bf1c98cdd3#diff-246ef5684820c36fd431bbc01589168fR83 
This line appears to be the culprit

**Comment by ShachafGoldstein** [comment]
I'm seeing something else, line 122 doesn't have a null check for and causes a null exception, and when the path doesn't exists I get just the HKU back from the line you mentioned

**Comment by mgoleary** [comment]
I'm getting the same error (as shown in Actual Results) when testing against both a path or property key that exists and also a path the doesn't exist or a property key that doesn't exist. If I add the colon after HKU on this line https://github.com/ansible/ansible/commit/e41312db3ca163a29609e1806f2e14bf1c98cdd3#diff-246ef5684820c36fd431bbc01589168fR83, everything works as expected

**Comment by jborean93** [comment]
Thanks @mgoleary for the report and identifying the actual issue. I've opened a PR to fix this https://github.com/ansible/ansible/pull/59359. The reason for the null exception is because `HKU` is not matched on the path qualifier so `$registry_hive` is set to `$null` which does not have the `.Dispose()` method. Looks like I found this issue in win_regedit before 2.8 was release so that is currently unaffected.
